R/mod_body_coc_competition.R

```diff
@@ -0,0 +1,475 @@
+#' body_coc_competition UI Function
+#'
+#' @description A shiny Module.
+#'
+#' @param id,input,output,session Internal parameters for {shiny}.
+#'
+#' @noRd 
+#'
+#' @importFrom shiny NS tagList 
+mod_body_coc_competition_ui <- function(id){
+  ns <- NS(id)
+  tagList(
+    ui_picker_program(
+          inputId = ns("pe_provider"),
+          label = "Select your CoC-funded Provider",
+          choices = sort(pe_summary_validation()$AltProjectName) |> unique(),
```

Comment:
A general fix to be applied throughout this PR:
the accessor functions, like `pe_summary_validation`, load a file from disk each time they are called. Each time this is called it adds compute time. 
The best way to reduce the compute time for these is to load it a single time within an environment and assign it to an object such that it exists in RAM to be used in each subsequent reference.

In this case, you'd want to add 
`pe_sum_val <- pe_summary_validation()` somewhere above `tagList` and then use it in subsequent calls. 

You can extrapolate this pattern to all the other places in the PR where accessor functions are called repeatedly.

Does that make sense?

Reply:
Yeah, that makes sense. For the other accessor functions below, should I put the new lines where I assign the object by pe_sum_val or should I put those inside the server (below the ns line)?